include/circt/Dialect/FIRRTL/FIRRTLUtils.h

```diff
@@ -138,11 +138,11 @@ getInnerRefTo(FModuleLike mod, size_t portIdx, StringRef nameHint,
 
 /// If it is a base type, return it as is. If reftype, return wrapped base type.
 /// Otherwise, return null.
-inline FIRRTLBaseType getBaseType(FIRRTLType type) {
-  return TypeSwitch<FIRRTLType, FIRRTLBaseType>(type)
+inline FIRRTLBaseType getBaseType(Type type) {
+  return TypeSwitch<Type, FIRRTLBaseType>(type)
```

Comment:
Should this support a null type? Nice to be able to assume input is not null, but just curious re:how these might chain/be used.

Reply:
I think traditionally the `dyn_cast_or_null` function can handle a null argument, and we have `getBaseTypeOrNull` which I guess should be updated to null-check and then forward to this function.  Also to make the name clearer we should rename this to follow the new upstream name for this `dyn_cast_if_present`, i.e. `getBaseTypeIfPresent`.

Reply:
https://github.com/llvm/circt/pull/5393